static/images/diagrams/cloud_ssh_workflow.txt

```diff
@@ -0,0 +1,41 @@
+# Use http://sequencediagram.org/ to generate image
+
+title Cloud-based provisioning with Puppet over ssh
+
+participant User
+participant "Provisioned Instance" as Host
+participant "Foreman" as Foreman
+participant "Foreman Proxy" as Proxy
+participant "Infrastructure Cloud" as Cloud
+participant DNS
+participant "Puppet master" as Puppet
+
+linear
+autonumber
+
+note over Foreman:has an Image with credentials defined
+User->Foreman:clicks on New Host
+User->Foreman:selects cloud resource, the image and submits
+Foreman->Proxy:creates DNS records
+Proxy->DNS:forwards DNS records
+Foreman->Proxy:creates DNS records
+Foreman->Cloud:creates new instance
+Cloud->Foreman:reports IP address
+Foreman->Cloud:starts the instance
+note over Host:boots up
+Foreman->Host:executes finish script via ssh
+
+== Finish script ==
+
+Host->Puppet:signs puppet client certificate (puppet agent)
+Host->Puppet:sends initial facts
+Puppet->Foreman:forwards initial facts
+Host->Foreman:turns off build mode
```

Comment:
Foreman turns off build mode itself, the host doesn't need to do anything.  (It happens at the end of the SSH finish script's execution.)